packages/wast-parser/src/grammar.js

```diff
@@ -811,20 +811,15 @@ export function parse(tokensList: Array<Object>, source: string): Program {
 
       while (token.type !== tokens.closeParen) {
         moduleFields.push(walk());
-
-        if (state.registredExportedElements.length > 0) {
-          state.registredExportedElements.forEach(decl => {
-            moduleFields.push(t.moduleExport(decl.name, decl.type, decl.id));
-          });
-
-          state.registredExportedElements = [];
-        }
-
```

Comment:
I think this was to make the items in `registredExportedElements` available for the next iteration of `walk`. If the tests are fine i'm fine too :+1: